platform/ext/target/nordic_nrf/common/cmsis_drivers/Driver_USART.c

```diff
@@ -0,0 +1,363 @@
+    return NRFX_SUCCESS;
+}
+
+int32_t ARM_USART_Transfer(const void *data_out, void *data_in, uint32_t num)
+{
+    ARG_UNUSED(data_out);
+    ARG_UNUSED(data_in);
+    ARG_UNUSED(num);
+
+    return ARM_DRIVER_ERROR_UNSUPPORTED;
+}
+
+uint32_t ARM_USARTx_GetTxCount(const UARTx_Resources *uart_resources)
+{
+    return uart_resources->tx_count;
+}
+
+uint32_t ARM_USARTx_GetRxCount(const UARTx_Resources *uart_resources)
+{
+    return uart_resources->rx_count;
+}
+
+int32_t ARM_USARTx_Control(uint32_t control, uint32_t arg,
+                           UARTx_Resources *uart_resources)
+{
+    if ((control & ARM_USART_CONTROL_Msk) != ARM_USART_MODE_ASYNCHRONOUS) {
+        return ARM_DRIVER_ERROR_UNSUPPORTED;
+    }
+
+    nrf_uarte_baudrate_t baudrate = uart_resources->baudrate;
+    nrf_uarte_config_t   hal_cfg  = uart_resources->hal_cfg;
+    switch (arg) {
+        case    1200: baudrate = NRF_UARTE_BAUDRATE_1200;    break;
+        case    2400: baudrate = NRF_UARTE_BAUDRATE_2400;    break;
+        case    4800: baudrate = NRF_UARTE_BAUDRATE_4800;    break;
+        case    9600: baudrate = NRF_UARTE_BAUDRATE_9600;    break;
+        case   14400: baudrate = NRF_UARTE_BAUDRATE_14400;   break;
+        case   19200: baudrate = NRF_UARTE_BAUDRATE_19200;   break;
+        case   28800: baudrate = NRF_UARTE_BAUDRATE_28800;   break;
+        case   31250: baudrate = NRF_UARTE_BAUDRATE_31250;   break;
+        case   38400: baudrate = NRF_UARTE_BAUDRATE_38400;   break;
+        case   56000: baudrate = NRF_UARTE_BAUDRATE_56000;   break;
+        case   57600: baudrate = NRF_UARTE_BAUDRATE_57600;   break;
+        case   76800: baudrate = NRF_UARTE_BAUDRATE_76800;   break;
+        case  115200: baudrate = NRF_UARTE_BAUDRATE_115200;  break;
+        case  230400: baudrate = NRF_UARTE_BAUDRATE_230400;  break;
+        case  250000: baudrate = NRF_UARTE_BAUDRATE_250000;  break;
+        case  460800: baudrate = NRF_UARTE_BAUDRATE_460800;  break;
+        case  921600: baudrate = NRF_UARTE_BAUDRATE_921600;  break;
+        case 1000000: baudrate = NRF_UARTE_BAUDRATE_1000000; break;
+        default:
+            return ARM_USART_ERROR_BAUDRATE;
+    }
+
+    if ((control & ARM_USART_DATA_BITS_Msk) != ARM_USART_DATA_BITS_8) {
+        return ARM_USART_ERROR_DATA_BITS;
+    }
+
+    switch (control & ARM_USART_STOP_BITS_Msk) {
+        case ARM_USART_STOP_BITS_1:
+            hal_cfg.stop = NRF_UARTE_STOP_ONE;
+            break;
+
+        case ARM_USART_STOP_BITS_2:
+            hal_cfg.stop = NRF_UARTE_STOP_TWO;
+            break;
+
+        default:
+            return ARM_USART_ERROR_STOP_BITS;
+    }
+
+    switch (control & ARM_USART_PARITY_Msk) {
+        case ARM_USART_PARITY_NONE:
+            hal_cfg.parity = NRF_UARTE_PARITY_EXCLUDED;
+            break;
+
+        case ARM_USART_PARITY_EVEN:
+            hal_cfg.parity = NRF_UARTE_PARITY_INCLUDED;
+            hal_cfg.paritytype = NRF_UARTE_PARITYTYPE_EVEN;
```

Comment:
I get a compile error here for nrf9160. Seems it doesn't have UARTE_CONFIG_PARITYTYPE_Msk.

Reply:
Right, I tested only on nRF5340. My bad. Will fix it promptly.

Reply:
Fixed.